rclrs_tests/src/parameter_service_tests.rs

```diff
@@ -0,0 +1,618 @@
+                assert_eq!(response.result.prefixes.len(), 0);
+            })
+            .unwrap();
+        try_until_timeout(|| *client_finished.read().unwrap())
+            .await
+            .unwrap();
+        *done.write().unwrap() = true;
+    });
+
+    res.await.unwrap();
+    rclrs_spin.await.unwrap();
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_get_set_parameters_service() -> Result<(), RclrsError> {
+    let context = Context::new([]).unwrap();
+    let (node, client) = construct_test_nodes(&context, "get_set");
+    let get_client = client.create_client::<GetParameters>("/get_set/node/get_parameters")?;
+    let set_client = client.create_client::<SetParameters>("/get_set/node/set_parameters")?;
+    let set_atomically_client = client
+        .create_client::<SetParametersAtomically>("/get_set/node/set_parameters_atomically")?;
+
+    try_until_timeout(|| {
+        get_client.service_is_ready().unwrap()
+            && set_client.service_is_ready().unwrap()
+            && set_atomically_client.service_is_ready().unwrap()
+    })
+    .await
+    .unwrap();
+
+    let done = Arc::new(RwLock::new(false));
+
+    let inner_node = node.node.clone();
+    let inner_done = done.clone();
+    let rclrs_spin = tokio::task::spawn(async move {
+        try_until_timeout(|| {
+            rclrs::spin_once(inner_node.clone(), Some(std::time::Duration::ZERO)).ok();
+            rclrs::spin_once(client.clone(), Some(std::time::Duration::ZERO)).ok();
+            *inner_done.read().unwrap()
+        })
+        .await
+        .unwrap();
+    });
+
+    let res = tokio::task::spawn(async move {
+        // Get an existing parameter
+        let request = GetParameters_Request {
+            names: seq!["bool".into()],
+        };
+        let client_finished = Arc::new(RwLock::new(false));
+        let call_done = client_finished.clone();
+        get_client
+            .async_send_request_with_callback(&request, move |response: GetParameters_Response| {
+                *call_done.write().unwrap() = true;
+                assert_eq!(response.values.len(), 1);
+                let param = &response.values[0];
+                assert_eq!(param.type_, ParameterType::PARAMETER_BOOL);
+                assert!(param.bool_value);
+            })
+            .unwrap();
+        try_until_timeout(|| *client_finished.read().unwrap())
+            .await
+            .unwrap();
+
+        // Getting both existing and non existing parameters should fail
+        let request = GetParameters_Request {
+            names: seq!["bool".into(), "non_existing".into()],
+        };
+        let client_finished = Arc::new(RwLock::new(false));
+        let call_done = client_finished.clone();
+        get_client
+            .async_send_request_with_callback(&request, move |response: GetParameters_Response| {
+                *call_done.write().unwrap() = true;
+                assert_eq!(response.values.len(), 0);
```

Comment:
As mentioned in earlier comments, this test doesn't match up with the ROS 2 documented expectations of the service, which is that the response always has the exact same size sequence as the request.